tests/test_core.py

```diff
@@ -447,6 +447,19 @@ def test_filter_variable_and_depth(test_df, filter_args):
     assert obs == []
 
 
+def test_filter_measurand_list(test_df):
+    data = test_df.data
+    data.loc[3, "variable"] = "foo"
+    data.loc[4, "unit"] = "bar"
+    df = IamDataFrame(data)
+
+    obs = df.filter(measurand=(("foo", "EJ/yr"), ("Primary Energy", "bar")))
+
+    assert obs.variable == ["foo", "Primary Energy"]
```

Comment:
Comparisons where unintentionally the order matters are always prone to breaking, I'd change it to:
```suggestion
    assert set(obs.variable) == {"foo", "Primary Energy"}
```